core/src/main/clojure/xtdb/expression/temporal.clj

```diff
@@ -460,6 +460,107 @@
                        (Duration/ofNanos)
                        (duration->mdn-interval)))})
 
+(defmethod expr/codegen-cast [:int :interval] [{{:keys [start-field end-field]} :cast-opts}]
+  (when end-field (throw (err/illegal-arg :xtdb.expression/attempting-to-cast-int-to-multi-field-interval
+                                          {::err/message "Cannot cast integer to a multi field interval"
+                                           :start-field start-field
+                                           :end-field end-field})))
+  {:return-type (if (#{"YEAR" "MONTH"} start-field)
+                  [:interval :year-month]
+                  [:interval :month-day-nano])
+   :->call-code (fn [[x]]
```

Comment:
Could probably call `single-field-interval` here, though I re-implemented such that we'd return month-day-nano. Likely one to update when we take a look at how we should handle single field interval and the like within plan.